library/importance.py

```diff
@@ -0,0 +1,105 @@
+from library.employee import Employee
+from library.neo4j import Neo4j
+from library.weaviate import Weaviate
+
+
+T = TypeVar('T')
+
+class ImportanceServiceException(Exception):
+    pass
+
+class ImportanceService:
+
+    _graph: Neo4j = None
+    @property
+    def graph(self):
+        if self._graph is None:
+            self._graph = Neo4j()
+        return self._graph
+    
+    _vector: Weaviate = None
+    @property
+    def vector(self):
+        if self._vector is None:
+            self._vector = Weaviate()
+        return self._vector
+    
+    def __init__(self):
+        dotenv.load_dotenv(dotenv_path=os.path.join(os.path.dirname(__file__), '../.env'))
+        self.decay_document_days = float(os.getenv('DECAY_DOC_DAYS', '7'))
+        self.decay_email_days = float(os.getenv('DECAY_EMAIL_DAYS', '7'))
+        self.decay_slack_days = float(os.getenv('DECAY_SLACK_DAYS', '7'))
+
+    decay_email_days: float = 7
+    decay_slack_days: float = 7
+    decay_document_days: float = 7
+
+    def add_importance_to_meeting(self, meeting: MeetingContext) -> None:
+        full_org = self.graph.get_employee_with_full_org_chart(meeting.person.email)
+
+        for item in meeting.support.docs:
+            document: DocumentResponse = self.vector.get_document_by_id(item.document_id)
+            owners: list[DocumentOwner] = document.metadata.owners
+            item.importance = self.decay_importance(ImportanceService._importance_document(full_org, owners), item.metadata.last_response, self.decay_document_days)
+
+        for item in meeting.support.email:
+            emails: list[EmailMessage] = self.vector.get_email_metadatas_in_thread(item.thread_id)
+            item.importance = self.decay_importance(ImportanceService._importance_email(full_org, emails), item.last_response, self.decay_email_days)
+
+        for item in meeting.support.slack:
+            slack_messages: list[SlackMessage] = self.vector.get_slack_thread_messages_by_id(item.thread_id).messages
+            item.importance = self.decay_importance(ImportanceService._importance_slack(full_org, slack_messages), item.last_response, self.decay_slack_days)
+    
+    def decay_importance(self, importance: float, last_response: datetime, decay_days: float) -> float:
+        delta = datetime.now().astimezone() - last_response
+        return importance * exp(-delta.days/decay_days)
+    
+    @staticmethod
+    def _identity_importance(org: Employee, email: str) -> int:
+        distance = org.distance_to(email)
+        importance = 0
+        if distance is None or distance == 0:
+            return 0
+        
+        if distance == 1 or distance == -1:
+            importance = 100
+        elif distance > 0:
+            importance = min(100, 25*distance)
+        else: # distance < 0
+            importance = max(0, 100 - 25*abs(distance))
+        return importance
```

Comment:
I was able to understand the distance == 1 or -1 case, but didn't quite understand how the importance score was calculated for skip managers and skip reportees.

Reply:
It's basically arbitrary logic, which is why I didn't put too much testing on it. 

Your manager two above is generally not communicating with you directly, mostly just departmental announcements, so I considered it less important. But then as you get more senior it seemed weird not to consider communications important (though I rarely found anything from the CEO of the company to be useful 6 layers down). 

The number of other people on an email and the identity of those users will be features we'll need to pay attention to in the real decision tree (which needs to be an xgboost or something and we'll need to get some labeled training data). Like if there are 1000 recipients or it's part of some DL as opposed to a direct communication from someone more senior (I had a few exchanges with the CEO directly and with the CPO that were certainly high importance in my inbox).

Reply:
Thank you for the explanation. That makes a lot of sense. I think the number of people on an email really has a huge role to play in importance.